### ISSUE TITLE
(?) method should not return false for string types

### ISSUE THREAD

**Issue by bobbytables** [cgriego/active_attr#111]
I realize we're following the YAML specification for boolean values. But I have a slight disagreement with that.

https://github.com/cgriego/active_attr/blob/master/lib/active_attr/typecasting/boolean_typecaster.rb#L18

In our case we had a value actually set to "F". So when we caled `attribute?` it returned false. When truly it wasn't.

I think the functionality should return false if nil, or if the type is boolean and false.

Basically, only initiate the BooleanTypecaster when it's a boolean type.
